include/RAJA/util/SubView.hpp

```diff
@@ -0,0 +1,268 @@
+        return 1;
+    }
+
+};
+
+template<typename IndexType = Index_type>
+struct NoSlice { 
+    static constexpr bool reduces_dimension = false;
+
+    RAJA_INLINE RAJA_HOST_DEVICE constexpr IndexType map_index(const IndexType& idx) const {
+        return idx;
+    }
+
+    template<IndexType DIM, typename LayoutType>
+    RAJA_INLINE RAJA_HOST_DEVICE constexpr IndexType size(const LayoutType& layout) const {
+        return layout.template get_dim_size<DIM>();
+    }
+
+    RAJA_INLINE RAJA_HOST_DEVICE constexpr IndexType stride() const {
+        return 1;
+    }
+};
+
+template<typename IndexType = Index_type>
+struct StridedSlice { 
+    IndexType start_, end_, stride_;
+
+    static constexpr bool reduces_dimension = false;
+
+    RAJA_INLINE RAJA_HOST_DEVICE constexpr IndexType map_index(const IndexType& idx) const {
+        return start_ + stride_ * idx;
+    }
+
+    template<IndexType DIM, typename LayoutType>
+    RAJA_INLINE RAJA_HOST_DEVICE constexpr IndexType size(const LayoutType&) const {
+        return (end_ - start_) / stride_ + 1;
+    }
+
+    RAJA_INLINE RAJA_HOST_DEVICE constexpr IndexType stride() const {
+        return stride_;
+    }
+};
+
+template<typename IndexType, typename... Slices>
+RAJA_INLINE RAJA_HOST_DEVICE constexpr auto make_slice_to_parent_index_map() {
+    IndexType sub_idx = 0;
+    IndexType i = 0;
+    camp::array<IndexType, sizeof...(Slices)> map = {};
+    ((map[i++] = (Slices::reduces_dimension ? -1 : sub_idx++)), ...);
+    return map;
+}
+
+template<typename IndexType, size_t n_parent_dims, typename... Slices>
+RAJA_INLINE RAJA_HOST_DEVICE constexpr auto make_parent_to_slice_index_map() {
+    IndexType sub_idx = 0;
+    IndexType i = 0;
+    camp::array<IndexType, n_parent_dims> map = {};
+
+    auto process_slice = [&](auto slice_type) constexpr {
+        if constexpr (!decltype(slice_type)::reduces_dimension) {
+            map[sub_idx++] = i++;
+        } else {
+            i++;
+        }
+    };
+
+    (process_slice(Slices{}), ...);
+
+    return map;
+}
+
+template <typename LayoutType, typename SliceTypes, typename IndexType = Index_type>
+struct SubRegion;
+
+/* SubLayout is a semantic alias for a SubRegion whose parent is a layout */
+template <typename LayoutType, typename SliceTypes, typename IndexType = Index_type>
+using SubLayout = SubRegion<LayoutType, SliceTypes, IndexType>;
+
+/* SubView is a semantic alias for a SubRegion whose parent is a view */
+template <typename LayoutType, typename SliceTypes, typename IndexType = Index_type>
+using SubView = SubRegion<LayoutType, SliceTypes, IndexType>;
+
+template <typename LayoutType, typename IndexType, typename... Slices>
+struct SubRegion<LayoutType, camp::list<Slices...>, IndexType> {
+
+    using IndexLinear = IndexType;
+
+    const LayoutType& parent_;
+    camp::tuple<Slices...> slices_;
+
+    static inline constexpr size_t num_slices_ = sizeof...(Slices);
```

Comment:
Static data members and functions should be above non-static data members and functions.